src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,106 @@
-import { useState } from 'react';
+/* eslint-disable max-len */
+import React, { useEffect, useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+interface Props {
+  onAdd: (movie: Movie) => void;
+}
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [count, setCount] = useState(0);
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [imgUrl, setImgUrl] = useState('');
+  const [imdbUrl, setImdbUrl] = useState('');
+  const [imdbId, setImdbId] = useState('');
+
+  const [disabledBtn, setDisabledBtn] = useState(true);
+
+  const validateUrl = (url: string) => {
+    const pattern = /^((([A-Za-z]{3,9}:(?:\/\/)?)(?:[-;:&=+$,\w]+@)?[A-Za-z0-9.-]+|(?:www\.|[-;:&=+$,\w]+@)[A-Za-z0-9.-]+)((?:\/[+~%/.\w-_]*)?\??(?:[-+=&;%@,.\w_]*)#?(?:[,.!/\\\w]*))?)$/;
+
+    return !pattern.test(url);
+  };
+
+  const deleteInputValues = () => {
+    setTitle('');
+    setDescription('');
+    setImgUrl('');
+    setImdbUrl('');
+    setImdbId('');
+  };
+
+  useEffect(() => {
+    if (title && imgUrl && imdbUrl && imdbId) {
+      setDisabledBtn(false);
+    } else {
+      setDisabledBtn(true);
+    }
+  }, [title, imgUrl, imdbUrl, imdbId]);
```

Comment:
```suggestion
const disabledBtn = !title.trim() || !imgUrl.trim() || !imdbUrl.trim() || !imdbId.trim()
```
it is redundant to manage state for such case. enough to make such check. trim every form field value should solve problem, when possible to add film with only spaces in each input (see screens)
![Screenshot 2024-01-13 at 17 45 14](https://github.com/mate-academy/react_movies-list-add-form/assets/33029864/a2548031-5231-4530-9c5e-d56c5baf014d)
![chrome-capture-2024-1-13 (1)](https://github.com/mate-academy/react_movies-list-add-form/assets/33029864/684ffd37-f36c-46e2-b942-f6d3eb774682)